### DISCUSSION TITLE
Examples of FrontDoor CDN Setup

### DISCUSSION THREAD

**Discussion by lelando** [brwilkinson/AzureDeploymentFramework#14] [Q&A] [answered]
Hi There,

I am looking through the modules here and I am wondering if you have an example of a frontdoor setup using CDN resources and not FrontDoor? I was really hoping to pass in an object with origins, routes, etc but finding that I am having trouble looping through an object with nested arrays and other objects. 

I see a CDN object being passed into `FD.CDN-Profiles.bicep` but not sure how it maps down to origin-groups, routes, etc.

Much appreciated!

**Comment by brwilkinson** [comment, marked as answer]
Sure, I have some updated templates and examples, that I can merge.

Will ping you back here, once I've done that.

**Reply by brwilkinson** [reply]
@lelando 

Here are the samples for the parameters

https://github.com/brwilkinson/AzureDeploymentFramework/blob/main/ADF/tenants/AOA/ACU1.T5.parameters.json#L793

plus the templates

https://github.com/brwilkinson/AzureDeploymentFramework/blob/main/ADF/bicep/FD.CDN.bicep
https://github.com/brwilkinson/AzureDeploymentFramework/blob/main/ADF/bicep/FD.CDN-Profiles.bicep
https://github.com/brwilkinson/AzureDeploymentFramework/blob/main/ADF/bicep/FD.CDN-Profiles-OrginGroups.bicep

**Reply by brwilkinson** [reply]
This is the example of how it maps out

![image](https://user-images.githubusercontent.com/3605266/174410816-8af547af-b3cd-4c86-9944-5733f4a03c61.png)

**Reply by lelando** [reply]
Thank you!

**Reply by brwilkinson** [reply]
I recently updated these templates.

Now it creates 1 or more profiles and then 1 or more afdendpoints and then creates 1 or more routes.

The default makes it easier to create a single AFDEndpoint and then add many routes and domains on that same shared afdendpoint.

**Reply by brwilkinson** [reply]
![image](https://user-images.githubusercontent.com/3605266/176513540-0091f4ba-80ec-481c-af64-8fd898fd6de1.png)
